### DISCUSSION TITLE
Can Convert to GGUF use multi-core?

### DISCUSSION THREAD

**Discussion by Taikono-Himazin** [ggml-org/llama.cpp#4262] [Q&A] [answered]
This is when using convert.py.

Is it possible to speed up the conversion from Hugging Face Model to GGUF by utilizing multi-cores?
I'm using MacStudio, but it has 24 cores, but only 1 core is used, so I feel like it's a waste.

There are mainly three things I would like to ask:
1. I would like to know if the feature is already implemented.
2. are there any plans to implement this feature?
3. I would like to ask if there is a reason why multi-core processing (parallel processing) is theoretically impossible in the first place.

Thank you!

**Comment by cmp-nct** [comment, marked as answer]
The conversion should mostly be limited in memory and disk speed, adding more cores is typically reducing performance not increasing it in these circumstances.
Quantizing is a step that is more compute intense and that tool already supports multi core processing

**Reply by Taikono-Himazin** [reply]
Oh, thank you. The usage of the disk was also not large, so I thought it was speeded by the CPU. But it was an SSD problem. I will think about replacing the SSD.
